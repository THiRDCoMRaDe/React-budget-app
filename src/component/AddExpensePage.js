import React from "react";
import { connect } from "react-redux";
//import {addExpense} from "../actions/expenses";
import {startAddExpense } from "../actions/expenses";
import ExpenseForm from './ExpenseForm';

export class AddExpensePage extends React.Component{
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };
    render() {
      return(
          <div>
              <div className="page-header">
                  <div className="content_container">
                    <h1 className="page-header__title">Add Expense</h1>
                  </div>
              </div>
              <div className="content_container">
                  <ExpenseForm
                      onsubmit={this.onSubmit}
                  />
              </div>
              {/*this is from my add expense component*/}
          </div>
      )
    }
}
const mapDispatchToProps = (dispatch) => ({startAddExpense: (expense) => dispatch(startAddExpense(expense)) });
export default connect(undefined, mapDispatchToProps)(AddExpensePage);