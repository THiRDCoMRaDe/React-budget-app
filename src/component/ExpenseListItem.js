import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({description, amount, createdAt, id}) => (
    <>
        <Link to={`/edit/${id}`}>
            <h3>
                description: {description}
            </h3>
        </Link>
        <p>
            {amount} - {createdAt}
        </p>

    </>
);


export default ExpenseListItem;
//export default ExpenseListItem;