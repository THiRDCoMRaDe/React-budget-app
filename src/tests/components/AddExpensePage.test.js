import React from 'react';
import { shallow } from 'enzyme';
import {AddExpensePage} from "../../component/AddExpensePage";
import Jest from 'jest-mock';
import {beforeEach, expect} from "@jest/globals";
import expenses from "../fixtures/expenses";

let addExpense, history, wrapper;
beforeEach(()=>{
    addExpense = Jest.fn();
    history = { push: Jest.fn() };
    wrapper = shallow(<AddExpensePage addExpense={addExpense} history={history} />);
});

test('should render AddExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onsubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(addExpense).toHaveBeenLastCalledWith(expenses[1]);
});