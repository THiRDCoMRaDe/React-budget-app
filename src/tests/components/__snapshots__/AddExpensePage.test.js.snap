// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddExpensePage correctly 1`] = `
<div>
  <h1>
    Add Expense
  </h1>
  <ExpenseForm
    onsubmit={[Function]}
  />
  this is from my add expense component
</div>
`;
